gpkit/constraints/set.py

```diff
@@ -44,6 +44,7 @@ def __init__(self, constraints, substitutions=None):
                         # so we can catch them (see above) in ConstraintSets
             if hasattr(self[i], "substitutions"):
                 self.substitutions.update(self[i].substitutions)
+                self[i].substitutions = self.substitutions
```

Comment:
I still think we don't have any good documentation of the behavior of substitutions for modular models with submodels. In particular , the fact that the submodel's substitutions get mutated to contain the larger model's subs is not obvious. I think this documentation should live all the way up at the read the docs level.